pkg/agent/config.go

```diff
@@ -258,151 +268,433 @@ func InitAgentCmdFlags(c *cobra.Command, cfg *AgentCmdFlags) {
+			if res.AuthMode == VenafiCloudKeypair {
+				// The VenafiCloudVenafiConnection mode doesn't need a server.
+				server = client.VenafiCloudProdURL
+			}
+		}
+		url, urlErr := url.Parse(server)
+		if urlErr != nil || url.Hostname() == "" {
+			errs = multierror.Append(errs, fmt.Errorf("server %q is not a valid URL", server))
 		}
+		if res.AuthMode == VenafiCloudVenafiConnection && server != "" {
+			log.Printf("ignoring the server field specified in the config file. In %s mode, this field is not needed.", VenafiCloudVenafiConnection)
+			server = ""
+		}
+		res.Server = server
+		res.EndpointPath = endpointPath
 	}
 
-	if flags.Period == 0 && cfg.Period == 0 && !flags.OneShot {
-		return Config{}, nil, fmt.Errorf("period must be set as a flag or in config")
-	}
+	// Validation of `venafi-cloud.upload_path`.
+	{
+		var uploadPath string
+		switch {
+		case res.AuthMode == VenafiCloudKeypair:
+			if cfg.VenafiCloud == nil || cfg.VenafiCloud.UploadPath == "" {
+				errs = multierror.Append(errs, fmt.Errorf("the venafi-cloud.upload_path field is required when using the %s mode", res.AuthMode))
+				break // Skip to the end of the switch statement.
+			}
+			_, urlErr := url.Parse(cfg.VenafiCloud.UploadPath)
+			if urlErr != nil {
+				errs = multierror.Append(errs, fmt.Errorf("upload_path is not a valid URL"))
+				break // Skip to the end of the switch statement.
+			}
 
-	var credentials client.Credentials
-	var err error
-	if flags.ClientID != "" {
-		credentials = &client.VenafiSvcAccountCredentials{
-			ClientID:       flags.ClientID,
-			PrivateKeyFile: flags.PrivateKeyPath,
+			uploadPath = cfg.VenafiCloud.UploadPath
+		case res.AuthMode == VenafiCloudVenafiConnection:
+			// The venafi-cloud.upload_path was initially meant to let users
+			// configure HTTP proxies, but it has never been used since HTTP
+			// proxies don't rewrite paths. Thus, we've disabled the ability to
+			// change this value with the new --venafi-connection flag, and this
+			// field is simply ignored.
+			if cfg.VenafiCloud != nil && cfg.VenafiCloud.UploadPath != "" {
+				log.Printf(`ignoring the venafi-cloud.upload_path field in the config file. In %s mode, this field is not needed.`, res.AuthMode)
+			}
+			uploadPath = ""
 		}
-	} else if flags.CredentialsPath != "" {
-		file, err := os.Open(flags.CredentialsPath)
-		if err != nil {
-			return Config{}, nil, fmt.Errorf("failed to load credentials from file %s: %w", flags.CredentialsPath, err)
+		res.UploadPath = uploadPath
+	}
+
+	// Validation of `uploader_id`.
+	//
+	// We found that `venafi-cloud.uploader_id` doesn't do anything in the
+	// backend. Since the backend requires it for historical reasons (but cannot
+	// be empty), we just ignore whatever the user has set in the config file,
+	// and set it to an arbitrary value in the client since it doesn't matter.
+	//
+	// TODO(mael): Remove the arbitrary `/no` path parameter from the Agent once
+	// https://venafi.atlassian.net/browse/VC-35385 is done.
```

Comment:
@inteon I added this comment to remind us that the `uploader_id` path parameter will soon be removed from the API (source: https://github.com/jetstack/jetstack-secure/pull/575#issuecomment-2363268458)